src/app/search/state/search.service.ts

```diff
@@ -86,7 +86,20 @@ export class SearchService {
    * @param attribute: workflow or tool property to sort by
    * @param direction: 'asc' or 'desc'
    */
-  compareAttributes(a: DockstoreTool | Workflow, b: DockstoreTool | Workflow, attribute: string, direction: SortDirection) {
+  compareAttributes(
+    a: DockstoreTool | Workflow,
+    b: DockstoreTool | Workflow,
+    attribute: string,
+    direction: SortDirection,
+    entryType: string
+  ) {
+    // For sorting tools by name, sort namespace property
+    // For sorting workflows by name, sort organization property
+    if (entryType === 'tool' && attribute === 'name') {
+      attribute = 'namespace';
```

Comment:
should be tool_path and full_workflow_path instead of namespace and organization (though I suppose that will all change once the index is redone)